swarm/api/http/server.go

```diff
@@ -640,7 +685,14 @@ func (s *Server) HandleGetFile(w http.ResponseWriter, r *Request) {
 	}
 
 	reader, contentType, status, err := s.api.Get(key, r.uri.Path)
+
 	if err != nil {
+		// cheeky, cheeky hack. See swarm/api/api.go:Api.Get() for an explanation
+		if rsrcErr, ok := err.(*api.ErrResourceReturn); ok {
```

Comment:
Rather than using this error, can't we just consult `contentType` here and call `handleGetResource` accordingly?

Reply:
Hmm I must have mixed up something, this code was not supposed to be in this PR.